src/Compilers/Core/AnalyzerDriver/AnalyzerManager.AnalyzerAndOptions.cs

```diff
@@ -18,7 +18,13 @@ public AnalyzerAndOptions(DiagnosticAnalyzer analyzer, AnalyzerOptions analyzerO
                 Debug.Assert(analyzerOptions != null);
 
                 Analyzer = analyzer;
-                _analyzerOptions = analyzerOptions;
+                // _analyzerOptions is used only as a key for equality testing, and only the properties
```

Comment:
@mavasani will this break analyzer that depends on doing var options = (WorkspaceAnalyzerOptions)context.analyzeroption?

...

this still leave the leak where AnalyzerAndOptions added to the map never gets removed.